src/app/ReadHandler.cpp

```diff
@@ -872,14 +876,21 @@ void ReadHandler::ForceDirtyState()
 
 void ReadHandler::SetStateFlag(ReadHandlerFlags aFlag, bool aValue)
 {
-    bool oldReportable = IsReportable();
+    bool oldReportable = ShouldStartReporting();
     mFlags.Set(aFlag, aValue);
 
     // If we became reportable, schedule a reporting run.
-    if (!oldReportable && IsReportable())
+    if (!oldReportable && ShouldStartReporting())
     {
-        // If we became reportable, the scheduler will schedule a run as soon as allowed
-        mObserver->OnBecameReportable(this);
+        if (IsActiveSubscription())
+        {
+            // If we became reportable, the scheduler will schedule a run as soon as allowed
+            mObserver->OnBecameReportable(this);
+        }
+        else if (ShouldReportUnscheduled())
```

Comment:
Can this ever happen?  The only state dependency in ShouldStartReporting is in the !ShouldReportUnscheduled case, no?

In particular, this "else if" block contradics the documentation on ShouldStartReporting.  That documention is the right thing, not this block.

Reply:
Definitely, this is a case of two solutions for one problem, the MoveToState needed that logic for a corner case and I applied it here too for consistency but it is no longer needed since of the check for !IsActiveSubscription in ShouldReportUnscheduled now takes care of that (explanation in other comment).

Will be removed.